src/keria/app/agenting.py

```diff
@@ -818,7 +818,7 @@ def on_post(self, req, rep):
 
         caid = icp.pre
 
-        if caid in self.agency.agents:
+        if caid in self.agency.agents or self.agency.get(caid=caid) is not None:
```

Comment:
Instead of the `or`, just call self.agency.get(caid=caid) as the sole if condition.  That method checks the `in` anyway.

Reply:
done.